README.md

```diff
@@ -0,0 +1,319 @@
+
+#### Configureable React
+
+TODO.
+
+- Explain component schemas
+
+### Providing blocks
+
+Let's create another app `partner.shelf` that provides an alternative shelf with a different React component. instead of providing a full template. The process is very similar to creating a template. We'll have both the `react` and the `store` builders. The folder structure might look something like:
+
+```txt
+.
+├── manifest.json
+├── react
+|   ├── package.json
+|   ├── Description.json
+|   └── Shelf.jsx
+└── store
+    └── blocks.json
+```
+
+The `manifest.json` declares both builders and the dependency on `vtex.shelf` and `vtex.product-summary`, since the app want to extend the Dream Store `shelf` block.
+
+```jsonc
+{
+    "vendor": "partner",
+    "name": "shelf",
+    "description": "A different shelf, made by partner.",
+    "builders": {
+        "store": "0.x",
+        "react": "2.x"
+    },
+    "dependencies": {
+        "vtex.product-summary": "1.x",
+        "vtex.shelf": "0.x"
+    }
+}
+```
+
+The `blocks.json` creates a `shelf` block that extends the `shelf` block on `vtex.shelf`. We could extend it more explicitly by calling it, for example, `shelf/alternative`, but, since we depend on someone who declares a `shelf`, it is clear that, by declaring another `shelf`, we are extending the one on `vtex.shelf`. We also add a new block inside the shelf, a `shelf-description` block, which loads a `Description` react component.
+
+```json
+{
+    "shelf": {
+        "name": "Shelf with description",
+        "component": "Shelf",
+        "blocks": ["shelf-description", "product-summary"]
+    },
+    "shelf-description": {
+        "component": "Description"
+    }
+}
+```
+
+This structure is enough to provide this alternative shelf to be used on the Storefront editor. When installed, this app will provide the alternative shelf.
+
+Note that we've also created a brand new block, that doesn't extend any previous declared block. The `shelf-description`. We can create brand new blocks all we want. We may, for example, create a `creative-block` that provides a brand new feature and add it to the home page by providing a new `store/home` that uses it.
+
+```json
+{
+    "store/home": {
+        "name": "Home page with a creative block",
+        "blocks": ["header/full", "banner", "shelf", "creative-block", "footer"]
+    ,
+    "creative-block": {
+        "component": "CreativeBlock"
+    }
+}
+```
+
+### Routes
+
+Suppose we want the `partner.theme` app to declare a new route. A new URL that can be accessed, providing some specific template. The `store` builder gives us that power via the `routes.json` file.
+
+```txt
+.
+├── manifest.json
+└── store
+    ├── routes.json
+    └── blocks.json
+```
+
+Suppose our store's url is `store.com` and we want the `store.com/custom` url to hit the `store/custom` template. We need to add a route to our `routes.json`.
+
+```jsonc
+{
+    "store/custom": {
```

Comment:
Would it be possible to just declare a route named `custom`? By naming it `store/custom`, the builder would check if it has a `header` and a `footer`, but I think there's a case where you'd want a landing page without `header` and `footer`.

Reply:
For now, we don't support.